packages/app-mobile/components/NoteEditor/CodeMirror/decorators.ts

```diff
@@ -0,0 +1,170 @@
+			const line = view.state.doc.lineAt(pos);
+			decorations.push({
+				pos: line.from,
+				decoration,
+			});
+
+			// Move to the next line
+			pos = line.to + 1;
+		}
+	};
+
+	const addDecorationToRange = (from: number, to: number, decoration: Decoration) => {
+		decorations.push({
+			pos: from,
+			length: to - from,
+			decoration,
+		});
+	};
+
+	for (const { from, to } of view.visibleRanges) {
+		ensureSyntaxTree(
+			view.state,
+			to
+		)?.iterate({
+			from, to,
+			enter: node => {
+				let blockDecorated = false;
+
+				// Compute the visible region of the node.
+				const viewFrom = Math.max(from, node.from);
+				const viewTo = Math.min(to, node.to);
+
+				switch (node.name) {
+				case 'FencedCode':
+				case 'CodeBlock':
+					addDecorationToLines(viewFrom, viewTo, codeBlockDecoration);
+					blockDecorated = true;
+					break;
+				case 'BlockMath':
+					addDecorationToLines(viewFrom, viewTo, mathBlockDecoration);
+					blockDecorated = true;
+					break;
+				case 'Blockquote':
+					addDecorationToLines(viewFrom, viewTo, blockQuoteDecoration);
+					blockDecorated = true;
+					break;
+				case 'InlineMath':
+					addDecorationToRange(viewFrom, viewTo, inlineMathDecoration);
+					break;
+				case 'InlineCode':
+					addDecorationToRange(viewFrom, viewTo, inlineCodeDecoration);
+					break;
+				case 'URL':
+					addDecorationToRange(viewFrom, viewTo, urlDecoration);
+					break;
+				case 'SetextHeading1':
+				case 'SetextHeading2':
+				case 'ATXHeading1':
+				case 'ATXHeading2':
+				case 'ATXHeading3':
+				case 'ATXHeading4':
+				case 'ATXHeading5':
+				case 'ATXHeading6':
+					addDecorationToLines(viewFrom, viewTo, headerLineDecoration);
+					break;
+				}
+
+				// Only block decorations will have differing first and last lines
+				if (blockDecorated) {
+					// Allow different styles for the first, last lines in a block.
+					if (viewFrom == node.from) {
+						addDecorationToLines(viewFrom, viewFrom, regionStartDecoration);
+					}
+
+					if (viewTo == node.to) {
+						addDecorationToLines(viewTo, viewTo, regionStopDecoration);
+					}
+				}
+			},
+		});
+	}
+
+	decorations.sort((a, b) => a.pos - b.pos);
+
+	// Items need to be added to a RangeSetBuilder in ascending order
+	const decorationBuilder = new RangeSetBuilder<Decoration>();
+	for (const { pos, length, decoration } of decorations) {
+		// Null length => entire line
+		decorationBuilder.add(pos, pos + (length ?? 0), decoration);
+	}
+	return decorationBuilder.finish();
+}
+
+const decoratorPlugin = ViewPlugin.fromClass(class {
+	public decorations: DecorationSet;
+
+	public constructor(view: EditorView) {
+		this.decorations = computeDecorations(view);
+	}
+
+	public update(viewUpdate: ViewUpdate) {
+		if (viewUpdate.docChanged || viewUpdate.viewportChanged) {
+			this.decorations = computeDecorations(viewUpdate.view);
+		}
+	}
+}, {
+	decorations: pluginVal => pluginVal.decorations,
+});
+
+export default decoratorPlugin;
```

Comment:
Shouldn't naming be consistent here? The file is called "decorator", it exports a "decoratorPlugin" variable, which is then imported as "decoratorExtension". Which one of these is the most accurate?

Reply:
Good point! The file should be named `decoratorExtension` and export `decoratorExtension`.